python/res/job_queue/queue.py

```diff
@@ -359,11 +396,69 @@ def execute_queue(self, pool_sema, evaluators):
             if evaluators is not None:
                 for func in evaluators:
                     func()
+            self.transition()
 
         if self.stopped:
             self.stop_jobs()
 
         self.assert_complete()
+        self.transition()
+
+    @staticmethod
+    def _translate_change_to_cloudevent(real_id, status):
+        return CloudEvent(
+            {
+                "type": _queue_state_event_type(status),
+                "source": f"/ert/ee/{0}/real/{real_id}/stage/{0}",
+                "datacontenttype": "application/json",
+            },
+            {
+                "queue_event_type": status,
+            },
+        )
+
+    @classmethod
+    async def _publish_changes(cls, changes, websocket):
+        events = [
+            cls._translate_change_to_cloudevent(real_id, status)
+            for real_id, status in changes.items()
+        ]
+        for event in events:
+            await websocket.send(to_json(event))
```

Comment:
This does not need to ba a class function I think
```suggestion
    @staticmethod
    async def _publish_changes(changes, websocket):
        events = [
            JobQueue._translate_change_to_cloudevent(real_id, status)
            for real_id, status in changes.items()
        ]
        for event in events:
            await websocket.send(to_json(event))
```